File: IT/1/Funzioni/Esercizi/Python/esercizio29.py
# file esercizio29.py

from __future__ import print_function
from sys import stdout
from os import linesep

def fibonacci(n):
	if (n <= 1):
		return n
	else:
		return fibonacci(n-1) + fibonacci(n-2)

for n in range(1, 21):
	print(fibonacci(n), end=" ")

stdout.write(linesep)

File: IT/1/Funzioni/Esercizi/C++/esercizio29.cpp
/* file esercizio29.cpp */

#include <iostream>

using std::cout;
using std::endl;

int fibonacci(int);

int main() 
{
	for (int i = 1; i <= 20; i++) {
		cout << fibonacci(i) << " ";
	}

	cout << endl;
}

int fibonacci(int n) {
	if (n <= 1) {
		return n;
	}
	else {
		return fibonacci(n-1) + fibonacci(n-2);
	}
}

